riscv/mmu.cc

```diff
@@ -395,17 +426,27 @@ reg_t mmu_t::walk(reg_t addr, access_type type, reg_t mode, bool virt, bool hlvx
       break;
     } else if (type == FETCH || hlvx ? !(pte & PTE_X) :
                type == LOAD          ? !(pte & PTE_R) && !(mxr && (pte & PTE_X)) :
-                                       !((pte & PTE_R) && (pte & PTE_W))) {
+                                       (type == STORE) && !((pte & PTE_R) && (pte & PTE_W))) {
       break;
     } else if ((ppn & ((reg_t(1) << ptshift) - 1)) != 0) {
       break;
     } else {
+      if (type & CBO) {
```

Comment:
This is unfortunate duplication of the XWR checking. Can't this be combined with the earlier block at line 427? Remove the `if (type & CBO)` then at the end `if (type == 0 || type == CBO) break;`.

Reply:
On second thought, my exception would break the exception cause because none of the `type` flags would be set anymore.

Reply:
OK. I'll try this way to combine them.